src/Microsoft.DotNet.CodeAnalysis/PackageFiles/codeAnalysis.targets

```diff
@@ -1,10 +1,19 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!-- 
+    We should be using $(BuildToolsTaskDir) for this, however we are currently forcing that property to 
+    always be net45 (because we use Mono to run our tests and Mono needs to load the net45 version)
+-->
+  <PropertyGroup>
+    <CodeAnalyzerDir Condition="'$(CodeAnalyzerDir)'=='' and '$(OSEnvironment)'=='Windows_NT'">$(ToolsDir)/net45/</CodeAnalyzerDir>
+    <CodeAnalyzerDir Condition="'$(CodeAnalyzerDir)'=='' and '$(OSEnvironment)'!='Windows_NT'">$(ToolsDir)/</CodeAnalyzerDir>
+  </PropertyGroup>
+  
   <ItemGroup Condition="'$(EnableDotnetAnalyzers)' == 'true'">
     <!-- Enable the Microsoft.Dotnet.CodeAnalysis.dll analyzer --> 
-    <Analyzer Include="$(BuildToolsTaskDir)Microsoft.Dotnet.CodeAnalysis.dll"/>
+    <Analyzer Include="$(CodeAnalyzerDir)analyzers/Microsoft.DotNet.CodeAnalysis.dll"/>
```

Comment:
Oh I see, looks like you are going to the analyzer in here, I think it would be better if your CodeAnalyzerDir already points to the analyzers dir instead of pointing to the tools dir and having to reference it here.